cinema/views.py

```diff
@@ -97,12 +100,27 @@ def get_queryset(self):
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
-
+        if self.action == "upload_image":
+            return MovieImageSerializer
         if self.action == "retrieve":
             return MovieDetailSerializer
-
         return MovieSerializer
 
+    @action(
+        methods=["POST"],
+        detail=True,
+        permission_classes=[IsAdminUser],
+        url_path="upload-image"
+    )
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Code Quality: You are currently returning a 400 status code when the serializer is not valid. It would be better to return a 422 status code (Unprocessable Entity) in this case. The 422 status code is used to indicate that the server understands the content type of the request entity, and the syntax of the request entity is correct, but it was unable to process the contained instructions.